src/adapter/etl-adapter-http/src/Flow/ETL/Adapter/Http/ContentTypeDetector.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Flow\ETL\Adapter\Http;
+
+use Psr\Http\Message\ResponseInterface;
+
+final class ContentTypeDetector
+{
+    public static function detectFromResponse(ResponseInterface $response) : string
+    {
+        foreach ($response->getHeader('Content-Type') as $header) {
+            if (\str_contains($header, 'application/json')) {
+                return 'json';
+            }
+
+            if (\str_contains($header, 'application/xml')) {
+                return 'xml';
+            }
+
+            if (\str_contains($header, 'text/html')) {
+                return 'html';
+            }
+        }
+
+        return 'text';
```

Comment:
last thing I don't like about this class is using string literals instead of enums as return type.

Reply:
Added a new enum.